### ISSUE TITLE
ARIA expanded pattern on sidebar navigation

### ISSUE THREAD

**Issue by srapilly** [facebook/docusaurus#9458]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [ ] I have tried the `npm run clear` or `yarn clear` command.
- [ ] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [ ] I have read the console error message carefully (if applicable).

### Description

The ARIA attribute `aria-expanded` is placed on each link that has a subsection

```html
<a class="menu__link menu__link--sublist menu__link--active" aria-current="page" aria-expanded="true" href="/docs/next/category/guides">Guides</a>
```

For example the **Guides** link on https://docusaurus.io/docs/next/category/guides

<img width="297" alt="" src="https://github.com/facebook/docusaurus/assets/85184231/621fd3e1-f071-433a-ae2a-be8d3bd754e5">

They should be on the `button` that collapses or expands the sublist

> The [aria-expanded](https://w3c.github.io/aria/#aria-expanded) attribute is applied to a focusable, interactive element that toggles visibility of content in another element. For example, it is applied to a parent [treeitem](https://w3c.github.io/aria/#treeitem) to indicate whether its child branch of the tree is shown. Similarly, it can be applied to a [button](https://w3c.github.io/aria/#button) that controls visibility of a section of page content.

### Steps to reproduce

1. Go to https://docusaurus.io/docs/next
2. Use a screen reader to navigate in the menu or the [accessibility tree can be visualized with chrome devtools](https://developer.chrome.com/blog/full-accessibility-tree/ )

### Expected behavior

Expected accessibility tree:

- link 'Advanced Guides' 
- button 'Advanced Guides' expanded: false or true

I think the accessible name of the button can also be shorter, the `aria-expanded` is already conveying the state, we don't need to keep "Collapse side category" or "Open sidebar category"

Related article: https://adrianroselli.com/2019/06/link-disclosure-widget-navigation.html 

### Actual behavior

Current accessibility tree:

<img width="582" alt="" src="https://github.com/facebook/docusaurus/assets/85184231/b95474eb-0950-470f-868c-b5986efc25f7">

- link 'Advanced Guides' expanded: false or true
- button 'Collapse sidebar category 'Advanced Guides'  

### Your environment

- Public site URL: https://docusaurus.io/docs/next

### Self-service

- [X] I'd be willing to fix this bug myself.

**Comment by ianjmacintosh** [comment]
It looks like this has been addressed.

I just checked on a local build and see the button element is now described accurately with the `aria-expanded` attribute:
```html
<div class="menu__list-item-collapsible">
  <a class="menu__link menu__link--sublist" href="/docs/advanced">Advanced Guides</a>
  <button aria-label="Expand sidebar category 'Advanced Guides'" aria-expanded="false" type="button" class="clean-btn menu__caret"></button>
</div>
```

Is this OK to close?

**Comment by srapilly** [comment]
@ianjmacintosh Nice! everything seems good now :)

**Comment by Josh-Cena** [comment]
For posterity: https://github.com/facebook/docusaurus/pull/9944
